packages/vm/src/state/baseStateManager.ts

```diff
@@ -480,7 +485,9 @@ export abstract class BaseStateManager {
     const accessList: AccessList = []
     folded.forEach((slots, addressStr) => {
       const address = Address.fromString(`0x${addressStr}`)
-      const check1 = getActivePrecompiles(this._common).find((a) => a.equals(address))
+      const check1 = getActivePrecompiles(this._common, this._customPrecompiles).has(
+        address.buf.toString('hex')
+      )
```

Comment:
This very much feels like the wrong abstraction to me to bake the precompiles even more into `StateManager`, I guess this was suboptimal from the beginning to have this `getActivePrecompiles` call in here.

This becomes even more apparent regarding the upcoming `StateManager` extraction.

Would it be a way to generally remove this `check1` and instead add the precompiles (e.g.) to the passed in `addressesRemoved` array? 🤔

Reply:
Ah right, I don't know why I added this, it should not be there. I think I had in the original PR also the remark that we should probably wait for the state manager to be extracted.